commands.py

```diff
@@ -0,0 +1,170 @@
+from nyuntam.settings import ROOT
+from nyuntam.utils.config import load_config, dump_config
+from nyuntam.constants.keys import FactoryArgumentKeys, JobServices
+
+# nyuntam_adapt
+from nyuntam_adapt.utils import AdaptParams, create_instance
+
+from dataclasses import dataclass, field, asdict
+from abc import ABC, abstractmethod
+from typing import List, Dict, Union, Optional
+import os
+from argparse import Namespace, ArgumentParser
+
+
+@dataclass
+class Command(ABC):
+    name: str = field(default="", init=False)
+    arguments: List[Union[str, "Command"]] = field(default_factory=list)
+    flags: Dict = field(default_factory=dict)
+
+    _command: Optional[str] = None
+
+    def build_command(self):
+        command = f"{self.name}"
+        for key, value in self.flags.items():
+            if value is not None:
+                command += f" --{key}={value}"
+        for arg in self.arguments:
+            if isinstance(arg, Command):
+                command += f" {arg.get_command_as_subcommand()}"
+            else:
+                command += f" {arg}"
+        return command
+
+    @property
+    def command(self):
+        if self._command is None:
+            self._command = self.build_command()
+        return self._command
+
+    def reset_command(self):
+        self._command = None
+
+    def reset_name(self):
+        self.name = ""
+        self.reset_command()
+
+    def run(self):
+        print(f"Running command: {self.command}")  ## remove
+        if os.system(self.command) != 0:
+            raise ValueError(f"Failed to run command: {self.command}")
+
+    @abstractmethod
+    def get_command_as_subcommand(self):
+        pass
+
+
+@dataclass
+class NyunRun(Command):
+    name = f"python {str(ROOT / 'main.py')}"
+
+    @classmethod
+    def from_namespace(cls, args: Namespace):
+        return cls(flags=vars(args))
+
+    def get_command_as_subcommand(self):
+        return self.command.replace("python", "").strip()
+
+
+@dataclass
+class NyunRunTorch(Command):
+    name = "torchrun"
+    flags = {"nnodes": 1, "nproc-per-node": 1}
+
+    @classmethod
+    def from_args(
+        cls,
+        args: Namespace,
+        num_gpu: Optional[int] = None,
+        num_nodes: Optional[int] = None,
+    ):
+        nyun_run = NyunRun.from_namespace(args)
+
+        num_nodes = num_nodes if num_nodes != None else cls.flags["nnodes"]
+        num_gpu = num_gpu if num_gpu != None else cls.flags["nproc-per-node"]
+        cls.flags.update({"nnodes": num_nodes, "nproc-per-node": num_gpu})
+        return cls(arguments=[nyun_run], flags=cls.flags)
+
+    def get_command_as_subcommand(self):
+        return self.command
```

Comment:
Because it wasn't being used. Why keep it ?

Reply:
makes sense. but in anyway we are passing the namespace and other arguments, defaulting to None, to this method, 🤔 should we not name this method itself as `from_namespace`; maintains consistency with other Command types that we have and makes more sense. why name `from_args` if we are not extending any parent class' method signatures?

Reply:
It is just a name. Will revert back to from_namespace. I didn't think that mattered much. And we discussed offline where you suggested to change the name form from_namespace to anything else. So, Since we are taking a few arguments here, I changed it to from_args

Reply:
nothing very wrong with from_args as well, except for slight inconsistency. I wouldn't have pointed it out if from_namespace was still there, but I didn't think we'll remove that method completely.

Reply:
It was removed because it was not being used. So, no point in having it. Will revert back to from_namespace.